File: interlayer/src/main/java/com/street/interlayer/controller/WalletController.java
package com.street.interlayer.controller;

import com.street.interlayer.model.Account;
import com.street.interlayer.model.Wallet;
import com.street.interlayer.service.AccountService;
import com.street.interlayer.service.WalletService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/wallet")
public class WalletController {

    @Autowired
    private WalletService walletService;

    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public ResponseEntity<Wallet> getWalletById(@PathVariable String id) {
        return ResponseEntity.ok(walletService.getWalletById(id));
    }

    @RequestMapping(value = "", method = RequestMethod.GET)
    public ResponseEntity<List<Wallet>> getAllWallets() {
        return ResponseEntity.ok(walletService.getAllWallets());
    }

    @RequestMapping(value = "", method = RequestMethod.POST)
    public ResponseEntity<Wallet> saveWallet(@RequestBody Wallet wallet) {
        if (wallet != null) {
            return ResponseEntity.ok(walletService.saveWallet(wallet));
        }
        return ResponseEntity.noContent().build();
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
    public void deleteBand(@PathVariable String id) {
        walletService.deleteWallet(id);
    }

}

File: micro/src/main/kotlin/com/street/micro/controller/WalletController.kt
package com.street.micro.controller

import com.street.micro.entity.Wallet
import com.street.micro.repository.WalletRepository
import com.street.micro.service.WalletService
import org.bson.types.ObjectId
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/wallet")
class WalletController(
    private val walletService: WalletService
) {
    @GetMapping
    fun getAllWallets(): ResponseEntity<List<Wallet>> {
        val wallets = walletService.getAllWallets()
        return ResponseEntity.ok(wallets)
    }

    @GetMapping("/{id}")
    fun getWalletById(@PathVariable("id") id: String): ResponseEntity<Wallet> {
        val wallet = walletService.getWalletById(id)
        return ResponseEntity.ok(wallet)
    }

    @PostMapping
    fun saveWallet(@RequestBody wallet: Wallet): Wallet {
        return walletService.saveWallet(wallet)
    }

    @DeleteMapping("/{id}")
    fun deleteWallet(@PathVariable id: String) {
        walletService.deleteWallet(id)
    }
}